packages/neuron-wallet/src/address/index.ts

```diff
@@ -0,0 +1,124 @@
+import TransactionsService from '../services/transactions'
+import ckbCore from '../core'
+import AddressType from './type'
+import HD from '../keys/hd'
+import { KeysData } from '../keys/keystore'
+
+const MaxAddressNumber = 30
+
+export interface HDAddress {
+  address: string
+  path: string
+}
+
+class Address {
+  public static isUsedAddress = (address: string) => {
+    return TransactionsService.hasTransactions(address)
+  }
+
+  public static addressFromPrivateKey = (privateKey: string) => {
+    const account = ckbCore.wallet.accountFromPrivateKey(Buffer.from(ckbCore.utils.hexToBytes(privateKey)))
+    return Address.addressFromPublicKey(account.hexPubKey)
+  }
+
+  public static addressFromPublicKey = (publicKey: string) => {
+    return ckbCore.utils.pubkeyToAddress(publicKey)
+  }
+
+  public static addressFromHDIndex = (keysData: KeysData, index: number, type = AddressType.Receiving) => {
+    const { privateKey } = HD.keyFromHDIndex(keysData, index, type)
+    return Address.addressFromPrivateKey(privateKey)
+  }
+
+  public static latestUnusedAddress = (keysData: KeysData) => {
+    const latestUnusedIndex = Address.searchAddress(keysData, 20)
+    const { privateKey } = HD.keyFromHDIndex(keysData, latestUnusedIndex, AddressType.Receiving)
+    return Address.addressFromPrivateKey(privateKey)
+  }
+
+  // Generate both receiving and change addresses
```

Comment:
It's weird that one is **doing** and one is **noun**

Reply:
Receiving can be as a noun.

Reply:
Here `receiving` and `change` are adj. and they are standard names from Bitcoin.